### PR TITLE
Bluetooth Host: HCI_RESERVE settings for G24-DK2601B

### PR DESCRIPTION
**Pull request by Dermiste**
Increase stack size to 4096 in case of shell usage
Set HCI_RESERVE to 1 if BT_SILABS_HCI is used

### CODE REVIEW

**Review by jori-nordic — DISMISSED** [review]

**Review by Thalley — APPROVED** [review]

**Review by fkokosinski — DISMISSED** [review]
Thanks for this :)

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -84,6 +84,7 @@ config BT_HCI_RESERVE
 	default 1 if BT_ESP32
 	default 0 if BT_B91
 	default 1 if BT_AMBIQ_HCI
+	default 1 if BT_SILABS_HCI
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig:87]: Is this change actually related to the shell? These values should relate to HCI-driver internal implementation details. If the silabs driver requires an extra byte, that's fine, but I think it would be cleaner to have that as a separate bug fix commit with a proper commit message explaining this (instead of mixing it together with a shell-related change).

Comment (Dermiste) [reply subsys/bluetooth/host/Kconfig:87]: It was required to make the BLE Shell operationnal on that board yes, I don't in depth knowledge of zephyr/ble to know where it's more practical to have it implemented

Comment (jhedberg) [reply subsys/bluetooth/host/Kconfig:87]: The one byte headroom requirement seems to come from here in the silabs HCI driver send() implementation:
https://github.com/zephyrproject-rtos/zephyr/blob/868193ef43e90b6f55753d8baa38701ef09ed8d9/drivers/bluetooth/hci/slz_hci.c#L98-L104

So it's an actual bug that indeed needed fixing. To be clear: I was only asking to split up your changes in this pull request to two separate commits, since the buffer headroom requirement has nothing to do with the shell (the bug could manifest itself in any Bluetooth use case with this HCI driver).

**Review by jhedberg — CHANGES_REQUESTED** [review]
See my inline comment. I'd prefer the bug fix to the driver to be clearly documented in a separate commit rather than being lost in something that on the surface seems only shell-related.

File: subsys/bluetooth/host/Kconfig (hunk 1 above)
Comment (Dermiste) [reply subsys/bluetooth/host/Kconfig:87]: Ok I'll make a separe PR. How does this work, should I commit and remove the amend to that line? Can the edit be made directly inside the PR ?

Comment (jhedberg) [reply subsys/bluetooth/host/Kconfig:87]: You could e.g. do it using something like:
```
git reset HEAD~2
git add boards/arm/efr32xg24_dk2601b/Kconfig.defconfig
git commit -v -s # Create the first commit message
git add subsys/bluetooth/host/Kconfig
git commit -v -s # Create the second commit message
git push --force <your PR branch in your fork>
```

Comment (Dermiste) [reply subsys/bluetooth/host/Kconfig:87]: Won't that be a problem for contribution guidelines? I thought I was supposed to make only a single squashed commit.

Comment (Thalley) [reply subsys/bluetooth/host/Kconfig:87]: > Won't that be a problem for contribution guidelines? I thought I was supposed to make only a single squashed commit.

The guidelines do not say that :) What you should avoid is to add commits that fixes your own commits based on e.g. review comments. You can have multiple commits in a PR when they e.g. change their own thing (like is the case here)

Comment (Dermiste) [reply subsys/bluetooth/host/Kconfig:87]: Done !

**Comment by Dermiste** [comment]
Is there anything more to be done on my side?

**Review by Thalley — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Comment by jhedberg** [comment]
> Is there anything more to be done on my side?

I don't think so, thanks for the updates! Now we just need to wait for the code freeze to be over to merge this.

**Comment by jori-nordic** [comment]
> code freeze to be over

Can we merge this @aescolar ?

**Comment by aescolar** [comment]
> Can we merge this @aescolar ?

This PR is gated by the merge of the the collab-hwmv2 branch which should happen tomorrow. Sorry for the inconvenience.

**Review by nordicjm — DISMISSED** [review]
Needs converting to hwmv2

**Comment by Dermiste** [comment]
> Needs converting to hwmv2

You're right it seems the path to this board `boards/arm/efr32xg24_dk2601b/ ` does not exist anymore, couldn't find the new location though, any tips?

Kconfig at `subsys/bluetooth/host/Kconfig` seems fine

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]
